homeassistant/components/axis/device.py

```diff
@@ -103,8 +103,7 @@ def name(self) -> str:
     @property
     def unique_id(self) -> str:
         """Return the unique ID (serial number) of this device."""
-        assert (unique_id := self.config_entry.unique_id)
-        return unique_id
+        return cast(str, self.config_entry.unique_id)
```

Comment:
But, this is a similar issue?

unique ID is allowed to be `None`.

```suggestion
        return self.config_entry.unique_id
```

and restore the return value of this method.

Reply:
`signal_reachable` property is the problem I guess. Why is that one using the config entry?

Reply:
It's actually the `DeviceInfo`. Both `identifiers` and `connections` expect a `set[tuple[str, str]]`.
That's quite a common issue AFAICT. I've usually defaulted to just adding `type: ignore[arg-type]` there.

Reply:
That isn't that common. Either is can be provided by the depending library or device already, and that is used, of the entry_id of the config entry is used.

I think the problem here, is that is should not be using the unique ID from the config entry.

Reply:
> That isn't that common. Either is can be provided by the depending library or device already, and that is used, of the entry_id of the config entry is used.
> 
> I think the problem here, is that is should not be using the unique ID from the config entry.

Im currently reworking the library so can redo that once the library and integration are in sync again.

Reply:
I don't think this should have been merged tbh.

Reply:
This PR?

Reply:
Yup, this and the previous one.

This should be handled correctly, without ignores, casts, or asserts:

Reply:
Sorry I thought the change was good enough and assumed you had resolved the comment. My comment on the library was not related to pushing through this change, just mentioning what I'm doing right now.

Reply:
Hello! I reworked it here https://github.com/home-assistant/core/pull/112132